File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        
        while start<=end:
            middle = start + (end - start)//2
            if nums[middle] == target:
                return middle
            elif nums[middle] < target:
                start = middle + 1
            else:
                end = middle - 1
                
        return -1

File: 0704-binary-search/0704-binary-search.c
int search(int* nums, int numsSize, int target){
    
    int l = 0;
    int r = numsSize - 1;
    int mid;
    
    while (l <= r) {
        mid = (r + l) / 2;
        if (nums[mid] < target) {
            l = mid + 1;
            
        } else if (nums[mid] > target) {
            r = mid - 1;
            
        } else {
            return mid;
        }
    }
    
    return -1;
}